### PR TITLE
Fix tools cross-client no-op

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug
Link: https://github.com/NuGet/Home/issues/5911
Regression: No  
If Regression then when did it last work:   
If Regression then how are we preventing it in future:   

## Fix
Details: Align the dg spec creation among both msbuild and vs tools restore to populate the same elements, add E2E tests  

## Testing/Validation
Tests Added: Yes
Reason for not adding tests:  
Validation done:  Tests + Manual validation

### CODE REVIEW

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/ToolRestoreUtility.cs [hunk 1]
```diff
@@ -52,16 +52,27 @@ public static PackageSpec GetSpec(string projectFilePath, string id, VersionRang
                     PackagesPath = packagesPath,
                     FallbackFolders = fallbackFolders,
                     Sources = sources,
+                    OriginalTargetFrameworks = {
+                        framework.GetShortFolderName()
```
Comment (rohit21agrawal) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/ToolRestoreUtility.cs]: don't call GetShortFolderName twice, it's not terribly expensive, but not cheap either.

**Review by rohit21agrawal — APPROVED** [review]
minor feedback, store short folder name. looks good overall

**Review by emgarten — APPROVED** [review]
LGTM, great test 🚀

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/ToolRestoreUtility.cs (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/ToolRestoreUtility.cs]: @rohit21agrawal good point, nice catch.

**Review by jainaashish — APPROVED** [review]
Looks good!

File: src/NuGet.Core/NuGet.Build.Tasks/GetRestoreDotnetCliToolsTask.cs [hunk 2]
```diff
@@ -117,7 +117,7 @@ public override bool Execute()
                 packageProperties.Add("TargetFrameworks", ToolFramework);
 
                 entries.Add(new TaskItem(Guid.NewGuid().ToString(), packageProperties));
-
+
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Build.Tasks/GetRestoreDotnetCliToolsTask.cs]: nit: extra space. please clean this.

**Review by mishra14 — APPROVED** [review]
nit comment. LGO!

**Pull request merged**